WebUI: Use thin scrollbars

skomerko opened the pull request:
Thin scrollbars are now used if they are supported by user's browser. The main goal was to make them less intrusive in Chrome-likes on some platforms.

a) Before:

![before](https://github.com/user-attachments/assets/89077e6f-669c-4d44-82da-20e979b568af)

b) After:

![after](https://github.com/user-attachments/assets/01a82952-c387-4623-a04c-c821de19c35c)

Chocobo1 commented on src/webui/www/private/css/style.css:
```diff
@@ -75,6 +75,18 @@ ol {
+
+@supports (scrollbar-width: auto) {
+    :root[slick-uniqueid], /* targets iframe mocha dialogs */
+    .panel,
+    .mochaContentWrapper,
+    .dynamicTableDiv,
```
I suppose these could be sorted?
```suggestion
    .dynamicTableDiv,
    .mochaContentWrapper,
    .panel,
```

thalieht commented:
Linux KDE
Tested in Brave and it's changed as advertised but in Firefox, the "After" is the normal and with this PR it becomes way too thin IMO. 
_However_ i don't remember if i changed anything about the scrollbar from the default in my Firefox profile.

And now that i mention it, like Firefox, i assume chrome based browsers have option(s) too to mess with the scroll bar's width. Won't this interfere with them?

skomerko commented:
> in Firefox, the "After" is the normal and with this PR it becomes way too thin IMO. _However_ i don't remember if i changed anything about the scrollbar from the default in my Firefox profile.

Here's the difference in Firefox 132, Win 11 24H2:
![before](https://github.com/user-attachments/assets/5c41c1d5-668d-4c40-bfe8-a0b77e36ecae) ![after](https://github.com/user-attachments/assets/c14d9996-b1ea-4132-b222-4e226b0a170e)

_thin_ scrollbar is platform dependend so the experience won't be the same in all circumstances. There are [accessibility](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-width#accessibility) concerns but there is keyboard scrolling too.

> And now that i mention it, like Firefox, i assume chrome based browsers have option(s) too to mess with the scroll bar's width. Won't this interfere with them?

It shouldn't. On Firefox I modified following options:

```
layout.css.scrollbar-width-thin.disabled

widget.non-native-theme.scrollbar.size.override
widget.non-native-theme.scrollbar.style
widget.non-native-theme.win.scrollbar.use-system-size
```

and they successfully overriden thin scrollbar. On Chrome I used Custom Scrollbar extension to create custom scrollbar and it worked too.

thalieht commented:
>Here's the difference in Firefox 132, Win 11 24H2:
![before](https://github.com/user-attachments/assets/5c41c1d5-668d-4c40-bfe8-a0b77e36ecae) ![after](https://github.com/user-attachments/assets/c14d9996-b1ea-4132-b222-4e226b0a170e)

The "before" looks the same for me but the "after" looks thinner:
![Screenshot_20241104_191323](https://github.com/user-attachments/assets/032aeb0f-cbae-4ae4-95f3-2c5eb4a10799) ![Screenshot_20241104_191546](https://github.com/user-attachments/assets/b246b196-9511-4638-bdc8-84870686df05)

>i don't remember if i changed anything about the scrollbar from the default in my Firefox profile.

Turns out i didn't touch anything.

Thankfully i don't use the WebUI much or this would bug me.

HanabishiRecca commented:
Scrollbars are adjustable in Firefox. Although, I'm not a fan of this change. Users **will** complain.

> The main goal was to make them less intrusive in Chrome-likes on some platforms.

Wouldn't it be better to go with `-webkit-scrollbar-*` selectors then? It only affects Chromium-based browsers and it's easy to make them look good.

skomerko commented:
> Scrollbars are adjustable in Firefox. Although, I'm not a fan of this change. Users **will** complain.
> 
> > The main goal was to make them less intrusive in Chrome-likes on some platforms.
> 
> Wouldn't it be better to go with `-webkit-scrollbar-*` selectors then? It only affects Chromium-based browsers and it's easy to make them look good.

I wanted to avoid creating anything custom for now, so I decided to simply extend current query to filter out Firefox - is it acceptable?

HanabishiRecca commented:
I didn't test it, but if that rule works as intended, then yes.
